src/mpid/ch4/netmod/include/netmod.h

```diff
@@ -514,6 +519,9 @@ typedef struct MPIDI_NM_funcs {
     MPIDI_NM_am_isend_reply_t am_isend_reply;
     MPIDI_NM_am_hdr_max_sz_t am_hdr_max_sz;
     MPIDI_NM_am_recv_t am_recv;
+    /* coll select function */
+    MPIDI_NM_algorithm_parser_t algorithm_parser;
+    MPIDI_NM_get_default_container_t get_default_container;
 } MPIDI_NM_funcs_t;
```

Comment:
As a note, we really need to discuss moving these out of device layer. The collection selection has no logical reason to be prefixed `NM`.

Reply:
We call `MPIDI_NM_algorithm_parser` and `MPIDI_NM_get_default_container` from CH4 and so I named it that way. Do you have any suggestions on renaming it?